helix-term/src/ui/editor.rs

```diff
@@ -316,16 +324,24 @@ impl EditorView {
         surface: &mut Surface,
         theme: &Theme,
         highlights: H,
+        whitespace: &helix_view::editor::WhitespaceConfig,
     ) {
+        use helix_view::editor::WhitespaceRenderValue;
+
         let text = doc.text().slice(..);
 
         let mut spans = Vec::new();
         let mut visual_x = 0u16;
         let mut line = 0u16;
         let tab_width = doc.tab_width();
-        let tab = " ".repeat(tab_width);
+        let tab = if whitespace.render.tab() == WhitespaceRenderValue::All {
+            (1..tab_width).fold(whitespace.substitutions.tab.to_string(), |s, _| s + " ")
```

Comment:
I think we might need to check the tab substitution width, if it's substituted with two-width characters then it will look twice as large.

Reply:
Is it possible to have a multi-column `char`? If so, we might be able to check for this when deserializing with `helix_core::graphemes::grapheme_width`